core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -228,7 +228,11 @@ public SelfT withKeyspace(String keyspaceName) {
    * @return a completion stage that completes with the session when it is fully initialized.
    */
   public CompletionStage<SessionT> buildAsync() {
-    return buildDefaultSessionAsync().thenApply(this::wrap);
+    CompletionStage<CqlSession> buildStage = buildDefaultSessionAsync();
+    CompletionStage<SessionT> wrapStage = buildStage.thenApply(this::wrap);
+    // thenApply does not propagate cancellation (!)
```

Comment:
Huh, did not know this, good to document 👍

Reply:
Yeah that was a big surprise for me, I'm pretty sure the Guava futures did.